Frontend/student-hub/src/pages/Profile.js

```diff
@@ -48,9 +47,31 @@ function Profile() {
         setError('Failed to fetch profile.');
       }
     };
+    // fetch unviersity list
+    const fetchUniversity = async () => {
+      try {
+        const { token } = getUserIDToken();
+        const response = await fetch (`${process.env.REACT_APP_API}/profile/university`, {
+          headers: {
+            'Authorization': `Bearer ${token}`
+          }
+        });
+        if (response.ok){
+          const data = await response.json();
+          setUniversities(data)
+        }else{
+          setError(`Failed to fetch universities: ${error.message}`);
```

Comment:
you can probably `throw` the error message instead, so that you catch the error in your catch block!